packages/plugin-ext/src/plugin/workspace.ts

```diff
@@ -106,6 +117,33 @@ export class WorkspaceExtImpl implements WorkspaceExt {
         this.workspaceFoldersChangedEmitter.fire(delta);
     }
 
+    $onTextSearchResult(result: SearchInWorkspaceResult, searchRequestId: number): void {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const isObject = (obj: any) => {
+            const type = typeof obj;
+            return type === 'function' || type === 'object' && !!obj;
+        };
+        result.matches.map(next => {
+            const range: Range = {
+                endColumn: next.character + next.length,
+                endLineNumber: next.line + 1,
+                startColumn: next.character,
+                startLineNumber: next.line + 1
+            };
+            const tRange = <theia.Range>Converter.toRange(range);
+            const searchResult: theia.TextSearchMatch = {
+                uri: URI.parse(result.fileUri),
+                preview: {
+                    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                    text: isObject(next.lineText) ? (<any>next.lineText).text : <string>next.lineText,
```

Comment:
Is this supposed to be a type-guard for `string | LinePreview`? If so, let's simplify to ` text: typeof next.lineText === 'string' ? next.lineText : next.lineText.text,` or use a proper type-guard and get rid of the `any`.